### PR TITLE
Assignment 2 - Synthesizing

### PR DESCRIPTION
**Pull request by smajid**
Completed this assignment. Ran into a lot of problems, read a lot, learnt a lot. This is just the beginning!
I sent a pull request, saw problems, made adjustments to the file, added and committed them, and now want to make sure everything is in order.

### CODE REVIEW

**Comment by smajid** [comment]
Professor, I checked the videos again, and tested my task03 file yet again. It passed both the tests.  But the problem, over here, persisted. Could you please let me know what am I doing wrong, after grading this assignment. I feel one error is the result of the previous files that I mistakenly sent pull request with, the ones you had to cancel earlier on. But I can clearly see task03 failing the test here. I hope you understand that I want to get better at this, that's why I want to know about this failure. Thanks!

**Comment by cheuschober** [comment]
@smajid, The test says that your value for the `Charlie` value is incorrect. It says what value it expects Charlie to have in the error message here:

http://128.228.53.195/job/is210-2015-spring-01-02-synthesizing/43/testReport/test_task_03/Task03TestCase/test_charlie/

Basically I wanted `Charlie` to be 'Brown`

**Comment by smajid** [comment]
Thanks, Prof.


I really was clueless on this one. Your input helped me, and I will be more careful about such potholes, on the way to a successful pull request. ?

_____________________

Saulat Majid
571.291.5913
________________________________

File: task_01.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ 1. Python can be used for almost any task you can think of.
+ 2. It has a huge library of available modules.
+ 3. Pip extends that library even more.
+ 4. Why aren't you using it already! Go write some Python!
+
+Sources
+
+====== ================
+Name   Contact
+====== ================
```
Comment (cheuschober) [inline task_01.rst]: You want to be careful about your alignment here. Python is *very* picky about spacing so make sure your table contents align.

Comment (smajid) [inline task_01.rst]: ?Sure. Thanks for the heads up, Professor.


_____________________

Saulat Majid
571.291.5913
________________________________

File: task_01.rst [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    me@mymachine $ python my_file_name.py
+
+
+**********
+The Syntax
+**********
+
+Python syntax is very easy to write. Take a look at this simple statement:
+
+.. code:: console
```
Comment (cheuschober) [inline task_01.rst]: Good! The only trick here is that this code isn't console, it's the python console so you could instead use:

    .. code:: pycon

To achieve a bit more consistent highlighting.

File: task_01.rst [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+====== ================
+Name   Contact
+====== ================
+Me          me@mysite.com
+Myself  myself@me.com
+ I               i@mii.com
+====== ================
+
+ +----------------------------------------------------------+
```
Comment (cheuschober) [inline task_01.rst]: While your solution here was VERY creative, the trick for this was a note directive, eg:

    .. note::

        This is just...

Comment (smajid) [inline task_01.rst]: ?Oh, alright. I didn't know that. I'll read more about it. It was very difficult, I kept trying many many options, finally it worked.



_____________________

Saulat Majid
571.291.5913
________________________________

**Comment by smajid** [comment]
Please ignore my 'heads up' comment, as I thought you were talking about Assignment#03's task, until I read the mail.


Also Professor, how would I know about using or even trying all these new options. I know they are there, but how do I look for them when I am stuck. I bank on the videos mostly, and if it's not there, I look up the reading material and if I can't find something there, I am stuck.


I am really disappointed in my work...both my warm up and synthesizing assignment didn't have a successful pull request.

?_____________________

Saulat Majid
571.291.5913
________________________________

**Comment by cheuschober** [comment]
Don't worry, you will get better, but the best way to do so I learning to love the internet. Learn the right keywords, google for something that looks or sounds like it might work, then try it and see if it got you closer. Reference resources like the docutils reference, python reference, and stackoverflow are all good places to search. The important part is losing the fear of just trying something. You can't *know* the answer up-front in most coding -- you just have to try and see what happens and much of that trial comes from educated guesses based on other, similar work adapted to your specific needs.

**Pull request merged**